velox/common/memory/Allocation.cpp

```diff
@@ -32,24 +32,10 @@ Allocation::~Allocation() {
 
 void Allocation::append(uint8_t* address, int32_t numPages) {
   numPages_ += numPages;
-  if (runs_.empty()) {
-    runs_.emplace_back(address, numPages);
-    return;
-  }
-
-  PageRun last = runs_.back();
```

Comment:
Just to confirm, the fix is to remove buggy optimization?

Reply:
yes.